modular_sand/code/game/objects/items/fleshlight.dm

```diff
@@ -137,22 +169,40 @@
 					for(var/obj/item/bodypart/l_arm/L in M.bodyparts)
 						can_interact = TRUE
 					if(can_interact)
-						message = (user == M) ? pick("touches softly against [src]") : pick("forces [M]'s finger on [src]")
+						if(portalunderwear.targetting == "vagina" || portalunderwear.targetting == "anus")
+							message = (user == M) ? pick("touches softly against [src]") : pick("forces [M]'s finger on [src]")
+						else if(portalunderwear.targetting == "penis")
+							message = (user == M) ? pick("jerks off [src]") : pick("forces [M] to jerk off [src]")
 						lust_amt = NORMAL_LUST
 						arouse_only_target = TRUE
 						target = "hand"
 	if(!useable)
 		to_chat(user, "<span class='notice'>It seems the device has failed or your partner is not wearing their device.</span>")
 	if(message)
 		var/mob/living/carbon/human/portal_target = ishuman(portalunderwear.loc) && portalunderwear.current_equipped_slot == ITEM_SLOT_UNDERWEAR ? portalunderwear.loc : null
+		switch(portalunderwear.targetting)
+			if("vagina")
+				V = portal_target.getorganslot(ORGAN_SLOT_VAGINA)
+			if("anus")
+				V = portal_target.getorganslot(ORGAN_SLOT_BUTT)
+			if("penis")
+				V = portal_target.getorganslot(ORGAN_SLOT_PENIS)
+		/*
 		if(portalunderwear.targetting == "vagina")
-			V = portal_target.getorganslot(ORGAN_SLOT_VAGINA)
+			V = portal_target.getorganslot(ORGAN_SLOT_VAGINA) // Why doesn't this have an equivalent for the anus?
```

Comment:
Because there's no anus organ. ORGAN_SLOT_BUTT contains the asscheeks.

Reply:
Gotcha, thank you for the heads up